src/extension.ts

```diff
@@ -519,30 +519,39 @@ class Device extends vscode.TreeItem {
     }
     
     async captureScreenshot() {
-        const error = (message) => {
-            vscode.window.showErrorMessage("Error capturing screenshot: " + message);
+        let anyErrors = false;
+        const handleCaptureError = (e) => {
+            anyErrors = true;
+            vscode.window.showErrorMessage("Error capturing screenshot: " + (e.message || e));
         }
-        
+
         try {
             const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
             const screenshotFile = `${screenshotDirectory}/ev3dev-${sanitizedDateString()}.png`;
 
             const conn = await this.exec('fbgrab -');
             const writeStream = fs.createWriteStream(screenshotFile);
+
+            conn.on('error', (e: Error) => handleCaptureError);
+
             writeStream.on('open', () => {
                 conn.stdout.pipe(writeStream);
             });
 
             writeStream.on('close', () => {
-                vscode.window.setStatusBarMessage(`Screenshot successfully captured`);
-                vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile));
+                if(!anyErrors) {

```

Comment:
> OK, what about destroying the stream or removing the 'finish' event handler when there is an error?

I can look into these alternatives.

> Also, I'm wondering... even if there is an error and we get to 'finish', then don't we have the entire file and it can be displayed?

If the client disconnects part-way through the download, we have half of a PNG file. That half of the PNG file is all of the data sent through the channel, but not all the data that `fbgrab` had to send.